tr069/src/main/java/com/github/freeacs/tr069/methods/INreq.java

```diff
@@ -213,8 +209,19 @@ public static void process(HTTPRequestResponseData reqRes, boolean isDiscoveryMo
         DBAccessSessionTR069 dbAccessSessionTR069 =
             new DBAccessSessionTR069(
                 reqRes.getDbAccess().getDBI().getAcs(), sessionData.getDbAccessSession());
+
+        String unitTypeName = deviceIdStruct.getProductClass();
+
+        if(properties.wantsAppendHwVersion())
+        {
+          String hardwareVersion = parser.getParameterList().getParameterValueByKey("Device.DeviceInfo.HardwareVersion");
```

Comment:
is hardware version static info? it can never change?

Reply:
Is different, when hardware revision changes. But I noticed in #12, some vendors (like ZTE) changes TR069 tree with new HW revisions. This PR works only in discovery mode, when UnitType is created dynamicly

Reply:
Ok, is this for testing, or is this actually needed in production?

Reply:
We need it in production, to test new version in huge scale. Currently > 25k devices. Maybe later I think to do this better...

Reply:
But if you can wait with merge, I want to check are this key "Device.DeviceInfo.HardwareVersion" is correct. I will check this today.

Reply:
if we are going to keep this, we need extensive documentation in the properties file that this will only work if discovery mode is active. And, we should check if discovery mode is active before append hw version. And log as warning if discovery mode is not active if the option is active.

Reply:
Ok

Reply:
This statement is in `if (isDiscoveryMode && sessionData.isFirstConnect())` block, lines 208

Reply:
ok. Can you add some comments in the property file that this will only work in discovery mode?

Reply:
Comments added, but I still didn't check if key is correct

Reply:
ahhh .... Device. You dont really know do you :) We have InternetGatewayDevice, which is old, and Device, that is new standard. We must support both. Let us think about this. There is some keyroot code in CPEParameters class. But I think ill have to look at this tomorrow.

Reply:
I used `sessionData.getKeyRoot()` to determine keyroot ;)